elastic/kibana
# [Logs UI]: Fix log error on fetching previous entries

## 📓 Summary

Closes #151700 

As shown in the video on [this comment](https://github.com/elastic/kibana/issues/151700#issuecomment-1497246729), the issue was reproduced when jumping to the lowest possible edge of the available logs in the lateral timeline.
In case there are no existing logs before the selected time cursor, the `topCursor` value required for the previous logs call is set to `null` and the error was triggered.

I believe in this case the conditional to trigger the error should not only rely on the missing `topCursor` variable existence, but we should also check whether previous logs exist with the `hasMoreBefore` state value.

These are the conditionals in order of execution:
- Top cursor missing but has more previous entries ➡️ trigger the Error
- No more previous entries and the request is not forced ➡️ Short circuit data fetching
- Top cursor exists => Trigger the request for previous entries.

These changes also spotted an existing bug where the `hasMoreBefore` was never set to the API response value, but was always `true`, preventing also showing the "Expand time range" call to action on the scenario described above.

## 🧪 Testing

- Navigate to Logs stream
- Jump to a point in time that has no logs previously ingested
- Verify the error is not logged when scrolling and that the "extend range" button is shown

https://user-images.githubusercontent.com/34506779/230292210-c2690c37-efa1-4b05-a237-f14eb78d26eb.mov
